### PR TITLE
Enumerables in Ruby

### PR DESCRIPTION
**Pull request by yaairfernando**
This change contains:
Enumerable ruby methods such as:
 **each
 **each_with_index
 **select
 **map
 **count
 **inject
 **all?
 **any?
 **none?
These methods were developed using a different logic other than using the actual ruby methods.

### CODE REVIEW

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      self_item.my_each { |x| y = false unless pattern == x }
+    elsif pattern.is_a? Regexp
+      self_item.my_each { |x| y = false unless x =~ pattern }
+    elsif pattern.is_a? Class
+      self_item.my_each { |x| y = false unless x.is_a? pattern }
+    else
+      self_item.my_each { |x| y = false unless x }
+    end
+    y
+  end
```
Comment (imhta) [inline enumerables.rb]: #### Failed test in `my_all?` method
- enumerables my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
- enumerables my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      self_item.my_each { |x| y = true if pattern == x }
+    elsif pattern.is_a? Regexp
+      self_item.my_each { |x| y = true if x =~ pattern }
+    elsif pattern.is_a? Class
+      self_item.my_each { |x| y = true if x.is_a? pattern }
+    else
+      self_item.my_each { |x| y = true if x }
+    end
+    y
+  end
```
Comment (imhta) [inline enumerables.rb]: #### Failed tests in `my_any?` method
-  enumerables my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
- enumerables my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex

File: enumerables.rb [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      self_item.my_each { |x| y = false if pattern == x }
+    elsif pattern.is_a? Regexp
+      self_item.my_each { |x| y = false if x =~ pattern }
+    elsif pattern.is_a? Class
+      self_item.my_each { |x| y = false if x.is_a? pattern }
+    else
+      self_item.my_each { |x| y = false if x }
+    end
+    y
+  end
```
Comment (imhta) [inline enumerables.rb]: #### Failed tests in `my_none?` method
- enumerables my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
- enumerables my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex

File: enumerables.rb [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    rst, sym = inj_param(*args)
+    arr = rst ? to_a : to_a[1..-1]
+    rst ||= to_a[0]
+    if block_given?
+      arr.my_each { |i| rst = yield(rst, i) }
+    elsif !sym.nil
+      arr.my_each { |i| rst = rst.public_send(sym, i) }
+    end
+    rst
+  end
```
Comment (imhta) [inline enumerables.rb]: #### Failed tests in my_inject method
- enumerables my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method

**Review by imhta — CHANGES_REQUESTED** [review]
# Changes required 🛑 

### Hello  @YairFernando67 , 👋 There some tests are failing for details check inline comment, fix 

----
#### Don't forgot to request for another code review
----

**Pon Muthu Selvam N**
_If you have any queries, feel free to ping me on slack_

**Comment by shshamim63** [comment]
## Status :green_apple:  [!! Approved !!] :green_apple: 
___
Hi @YairFernando67  :wave,
Awesome job :clap: . You have covered all the changes requested by the previous TSE. Just one case is falling behind. And that is really simple one. That why I am not blocking this project, but before merging it you have to cover it.
___
##### Please follow the instruction below :point_down: .
___
### :spades: [Mandatory Changes] :spades: 
___
1. ** my_all** -> If I execute [1, 2, 3].my_all? it should return me true and [1, false, 3].my_all? it should return false.
Which indicates that ,
#####  when no block or argument is given returns true when none of the collection members are false or nil
https://github.com/YairFernando67/Enumerables/blob/e94e4fb9095e2f3edb842ed1188b235567cb2b56/enumerables.rb#L41-L58
on your code, the first if check for block given or not. elsif checks for pattern is given. So the else should cover the above mentioned case.
instead of this :point_down: line, 
https://github.com/YairFernando67/Enumerables/blob/e94e4fb9095e2f3edb842ed1188b235567cb2b56/enumerables.rb#L55
I did `self_item.my_each { |x| y = false unless x }` and it worked.
I am sure that you just missed it accidentally :smile: .
___
#### Please make change on the above mentioned part and then you can merge it. (If you need more clarification don't  hesitate to knock me.)
___
## :herb:  Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/) | [Slack](@Shakhawat )

**Pull request merged**